### PR TITLE
core-bpf: special-case core bpf program error codes

### PR DESCRIPTION
**Pull request by buffalojoec**
This change special-cases a few error codes for the Address Lookup Table and Config programs, which otherwise would produce known mismatches between BPF and builtin.

For more information, see the comments above the change, or https://github.com/solana-program/config/pull/21.

### CODE REVIEW

**Comment by buffalojoec** [comment]
@mjain-jump feel free to review, but don't merge until we land on https://github.com/solana-program/config/pull/21 and https://github.com/solana-program/address-lookup-table/pull/43 please.

**Review by mjain-jump — APPROVED** [review]
looks good!

**Comment by buffalojoec** [comment]
@mjain-jump OK, those two linked PRs landed. PR is unchanged except for the removal of one code for each program. The programs only deal with read-only checks, not executable.

**Pull request merged**
